pkg/credentialprovider/aws/aws_credentials.go

```diff
@@ -36,6 +40,29 @@ const awsStandardDNSSuffix = "amazonaws.com"
 const awsChinaDNSSuffix = "amazonaws.com.cn"
 const registryURLTemplate = "*.dkr.ecr.%s.%s"
 
+func init() {
+	seen := sets.NewString()
+
+	for _, p := range endpoints.DefaultPartitions() {
+		for r, _ := range p.Regions() {
+			if !seen.Has(r) {
+				registerCredentialsProvider(r)
+				seen.Insert(r)
+			}
+		}
+	}
+
+	// ap-northeast-3 is purposely excluded from the SDK because it requires an
+	// access request see: https://github.com/aws/aws-sdk-go/issues/1863
+	registerCredentialsProvider("ap-northeast-3")
+
+	// Register the current region if we haven't seen it already just in case
+	// we're in a region that's not in the SDK for some reason.
+	if currentRegion, err := getCurrentRegion(); err == nil && !seen.Has(currentRegion) {
+		registerCredentialsProvider(currentRegion)
+	}
```

Comment:
I would `glog.Warningf` here if we can't get the region (`err != nil`)

Reply:
Updated